iohub/mm_fov.py

```diff
@@ -132,19 +132,34 @@ def hcs_position_labels(self):
                 labels = [
                     pos["Label"].split("-Pos") for pos in self.stage_positions
                 ]
+
                 # remove underscore from FOV name, i.e. '000_000'
                 # collect all wells in row '0' so output is
                 # ('0', '1', '000000')
                 return [
                     ("0", col, fov.replace("_", "")) for col, fov in labels
                 ]
             except Exception:
-                labels = [pos.get("Label") for pos in self.stage_positions]
-                raise ValueError(
-                    "HCS position labels are in the format of "
-                    "'A1-Site_0', 'H12-Site_1', or '1-Pos000_000' "
-                    f"Got labels {labels}"
-                )
+                try:
+                    # Look for "'Pos-1-000_000', 'Pos-2-000_001', ... "
+                    # and split into ('1', '000_000'), ...
+                    # New format following
+                    # https://github.com/micro-manager/micro-manager/pull/1897
+                    labels = [
+                        pos["Label"].split("Pos-")[1].split("-")
+                        for pos in self.stage_positions
+                    ]
+                    return [
+                        ("0", col, fov.replace("_", "")) for col, fov in labels
+                    ]
+
+                except Exception:
+                    labels = [pos.get("Label") for pos in self.stage_positions]
+                    raise ValueError(
```

Comment:
I feel that we have made this pile of error catching too big over time. At this point a regex match at the top and throwing early is probably a better solution.

Reply:
And by 'too big' I mean a 20-character indentation 😅